src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala

```diff
@@ -1069,4 +1029,184 @@ final case class EServicesApiServiceImpl(
       }
     }
   }
+
+  override def importEService(fileResource: FileResource)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem],
+    toEntityMarshallerCreatedResource: ToEntityMarshaller[CreatedResource]
+  ): Route = {
+
+    def extractZipToTempDirectory(zipBytes: Array[Byte], tenantId: String): Either[Throwable, Path] = {
+      Try {
+        val tempDir = Files.createTempDirectory(s"$tenantId/extractedZip") // Let me know if you have better suggestions
+        Using.resource(new ZipInputStream(new ByteArrayInputStream(zipBytes))) { zipIn =>
+          LazyList
+            .continually(zipIn.getNextEntry)
+            .takeWhile(_ != null)
+            .foreach { entry =>
+              val filePath = tempDir.resolve(entry.getName)
+              if (entry.isDirectory) {
+                Files.createDirectories(filePath)
+              } else {
+                Files.createDirectories(filePath.getParent)
+                Using.resource(new BufferedOutputStream(Files.newOutputStream(filePath))) { out =>
+                  val buffer = new Array[Byte](1024)
+                  Iterator
+                    .continually(zipIn.read(buffer))
+                    .takeWhile(_ != -1)
+                    .foreach(out.write(buffer, 0, _))
+                }
+              }
+            }
+        }
+        tempDir
+      }.toEither
+    }
+
+    def checkZipStructure(path: Path): Either[InvalidZipStructure, ImportedEservice] = {
+      val directoryName = path.getFileName.toString
+
+      def fileExists(filePath: Path): Either[InvalidZipStructure, Unit] = {
+        Either.cond(Files.exists(filePath), (), InvalidZipStructure(directoryName, s"File in '$filePath' not found"))
+      }
+
+      for {
+        jsonContent      <- Either
+          .catchNonFatal(Files.readString(path.resolve("configuration.json")))
+          .leftMap(ex => InvalidZipStructure(directoryName, "Error reading configuration.json: " + ex.getMessage))
+        importedEservice <- Try(jsonContent.parseJson.convertTo[ImportedEservice]) match {
+          case Success(eservice) => Right(eservice)
+          case Failure(ex)       =>
+            Left(InvalidZipStructure(directoryName, "Error decoding configuration.json: " + ex.getMessage))
+        }
```

Comment:
very minor, just an alternative
```suggestion
        importedEservice <- Try(jsonContent.parseJson.convertTo[ImportedEservice]).toEither.leftMap(ex =>
          InvalidZipStructure(directoryName, "Error decoding configuration.json: " + ex.getMessage)
        )
```